cdk/app/lib/consul-server.ts

```diff
@@ -1,34 +1,44 @@
 import * as fs from 'fs';
 import * as cdk from '@aws-cdk/core';
-import * as ec2 from "@aws-cdk/aws-ec2";
-import * as iam from "@aws-cdk/aws-iam";
-import { ServerInputProps } from './shared-props';
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as iam from '@aws-cdk/aws-iam';
+import * as secretsmanager from '@aws-cdk/aws-secretsmanager';
+import { ServerInputProps, ServerOutputProps } from './shared-props';
 
 export class ConsulServer extends cdk.Stack {
-  public readonly serverTag: {[key:string]: string};
-  public readonly datacenter: string;
+  public readonly props: ServerOutputProps;
 
   constructor(scope: cdk.Construct, id: string, inputProps: ServerInputProps) {
     super(scope, id, inputProps);
-    
+
     const ami = new ec2.AmazonLinuxImage({
       generation: ec2.AmazonLinuxGeneration.AMAZON_LINUX_2,
     });
 
+    const agentCASecret = new secretsmanager.Secret(this, 'agentCASecret', {
+      description: 'Consul TLS encryption CA public key'
+    });
+
+    const gossipKeySecret = new secretsmanager.Secret(this, 'gossipKeySecret', {
+      description: 'Consul gossip encryption key'
+    });
+
     // Role to allow Consul server to write to secrets manager
     const role = new iam.Role(this, 'ConsulSecretManagerRole', {
       assumedBy: new iam.ServicePrincipal('ec2.amazonaws.com'),
     });
     role.addToPolicy(new iam.PolicyStatement({
-      resources: [`arn:${cdk.Stack.of(this).partition}:secretsmanager:${cdk.Stack.of(this).region}:${cdk.Stack.of(this).account}:secret:*`],
-      actions: ['secretsmanager:CreateSecret'],
-      conditions: {"ForAnyValue:StringLike": {"secretsmanager:Name": ["my_consul-agent-ca*","my_consul-gossip-key*"]}},
+      actions: ['secretsmanager:UpdateSecret'],
+      resources: [agentCASecret.secretArn, gossipKeySecret.secretArn],
```

Comment:
It will be `secreteCompleteArn` now

Reply:
I changed `agentCASecret` and `gossipKeySecret` from `SecretAttributes` into `ISecret`. The reason is because The user-data.txt no longer create the secrets. Instead, I created the secrets via CDK (on consul-server.ts) and pass it as shared-props.

Reply:
But does secretArn still works in [here](https://github.com/aws-samples/amazon-ecs-fargate-consul-connect-example/pull/16/commits/a8f0b8f5a6c7e8acd1e530034a89534277ebf6f6#diff-6fd71a9717266395902d35ec296c5bfeabd03919d014d60b824e3ec658b0c78eR39) when using aws-cdk 1.133.0?

Reply:
Yes [ISecret](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-secretsmanager.ISecret.html) still uses `secretArn`, meanwhile it's deprecated on [SecretAttributes](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-secretsmanager.SecretAttributes.html)

Both `agentCASecret` and `gossipKeySecret` are no longer imported secrets btw, they are created and passed from consul-server.ts as shared properties.